### PR TITLE
#370 #377 #378 Builtin as default import and minor changes

### LINKED ISSUE #370: CI: sunset .jar release
I see we still build .jar in CI on release. I think we can only release `.js` build, it will make releases faster.

### LINKED ISSUE #377: Use custom logger or some lib for output instead of `println`
Theoretically, we can run into problems using `println` and we can customize styles of output with another approach

### LINKED ISSUE #378: Add builtins as default into `run`
This will prevent the need to constantly import builtins and copy them near every file with `aqua` code

### PR DESCRIPTION
**Pull request by DieMyst**
fixes #370 
fixes #377 
fixes #378

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -57,7 +59,8 @@ lazy val cli = crossProject(JSPlatform, JVMPlatform)
 
 lazy val cliJS = cli.js
   .settings(
-    scalaJSLinkerConfig             ~= (_.withModuleKind(ModuleKind.ESModule)),
+    scalaJSLinkerConfig ~= (_.withModuleKind(ModuleKind.ESModule)
+      .withESFeatures(_.withESVersion(ESVersion.ES2021))),
```
Comment (alari) [inline build.sbt:63]: Can you please leave a comment about why that's necessary?

File: cli/.js/src/main/scala/aqua/js/UtilsTypes.scala [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package aqua.js
+
+import scala.scalajs.js
+import scala.scalajs.js.annotation.{JSExportAll, JSImport}
+
+object Meta {
+
+  // get `import`.meta.url info from javascript
```
Comment (alari) [inline cli/.js/src/main/scala/aqua/js/UtilsTypes.scala:8]: Why do you want to get it?

File: cli/.js/src/main/scala/aqua/Utils.scala [hunk 3]
```diff
@@ -1,8 +1,13 @@
 package aqua
 
-import scribe.Level
+import aqua.js.{AvmLogLevel, FluenceJSLogLevel, Meta, Module}
+import fs2.io.file.Path
+import scribe.{Level, Logging}
+
+import scala.util.Try
+
+object Utils extends Logging {
```
Comment (alari) [inline cli/.js/src/main/scala/aqua/Utils.scala:9]: For me, any `Utils`-named class/object seems like a tech debt. It addresses what problem? How does it help?

File: cli/.js/src/main/scala/aqua/PlatformUtils.scala [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+import aqua.Utils.logger
+import aqua.js.{Meta, Module}
+import fs2.io.file.Path
+import scribe.Logging
+
+import scala.util.Try
+
+// JS-specific functions
+object PlatformUtils extends Logging {
```
Comment (alari) [inline cli/.js/src/main/scala/aqua/PlatformUtils.scala:11]: If it's js-specific, why `PlatformUtils`? What problem it solves? Is it for builtins? Is it possible to create smth like `BuiltinsSupport` and collect everything needed for builtins support there?

File: cli/.jvm/src/main/scala/aqua/PlatformUtils.scala [hunk 5]
```diff
@@ -0,0 +1,12 @@
+package aqua
+
+import fs2.io.file.Path
+import scribe.Logging
+
+import scala.util.Try
+
+// Java-specific functions
+object PlatformUtils extends Logging {
+  // you need to add import with builtin.aqua manually when running in JVM environment
```
Comment (alari) [inline cli/.jvm/src/main/scala/aqua/PlatformUtils.scala:10]: Are you adding builtins to `aqua run` (which is present only in JS and is completely meaningless in JVM environment) or to all the builds as automatic imports?

File: cli/.js/src/main/scala/aqua/run/RunCommand.scala [hunk 6]
```diff
@@ -77,7 +79,7 @@ object RunCommand extends Logging {
               PeerConfig(multiaddr, config.timeout, Utils.logLevelToAvm(config.logLevel), secretKey)
             )
             .toFuture
-          _ = println("Your peerId: " + peer.getStatus().peerId)
+          _ = OutputPrinter.print("Your peerId: " + peer.getStatus().peerId)
```
Comment (alari) [inline cli/.js/src/main/scala/aqua/run/RunCommand.scala:82]: Why the OutputPrinter is effectless? It's not pure, but it's used like it is

File: cli/src/main/scala/aqua/io/OutputPrinter.scala [hunk 7]
```diff
@@ -0,0 +1,13 @@
+package aqua.io
+
+// Uses to print outputs in CLI
+object OutputPrinter {
+
+  def print(str: String): Unit = {
```
Comment (alari) [inline cli/src/main/scala/aqua/io/OutputPrinter.scala:6]: It seems like an effect. Why just `Unit`?

**Review by alari — CHANGES_REQUESTED** [review]

**Comment by alari** [comment]
#378 is about adding builtins into `run`. It looks like you're adding builtins to all the calls, am I right? Why so?

It seems to me that smth like `Prelude` should be introduced. That `Prelude` should wrap all the default imports, including platform-specific ones. It should be possible to list the import filenames, and possibly import paths to the `Prelude` and have it configured.

For the JVM build, I think the Prelude should work the same way as for JS build, except not trying to add a new imports path for it.

For the `aqua run`, it seems like adding the global node_modules to imports (so that `import "@fluencelabs/aqua-lib/builtins.aqua"` should be pretty enough. Maybe some specific flag should be used to opt-in.

File: cli/.js/src/main/scala/aqua/Prelude.scala [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+import aqua.js.{Meta, Module}
+import fs2.io.file.Path
+import scribe.Logging
+
+import scala.util.Try
+
+/**
+ * @param pathList list of paths where imports will be searched
+ */
+case class Prelude(pathList: List[Path])
```
Comment (alari) [inline cli/.js/src/main/scala/aqua/Prelude.scala:13]: `importPaths`?

File: cli/.js/src/main/scala/aqua/Prelude.scala [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+
+// JS-specific functions
+object Prelude extends Logging {
+
+  def apply(): Prelude = {
```
Comment (alari) [inline cli/.js/src/main/scala/aqua/Prelude.scala:18]: Please not `apply`, smth else, or we will die trying to refactor it in the future

File: cli/.js/src/main/scala/aqua/js/UtilsTypes.scala (hunk 2 above)
Comment (alari) [reply cli/.js/src/main/scala/aqua/js/UtilsTypes.scala:8]: Please add comments (what does createRequire mean is also unclear) & rename `UtilsTypes` file to smth more meaningful

File: cli/src/main/scala/aqua/AppOpts.scala [hunk 10]
```diff
@@ -116,15 +121,15 @@ object AppOpts {
           }
         }
 
+        // TODO: move it to prelude after it will be used it in a compilation command, not only in `run`
```
Comment (alari) [inline cli/src/main/scala/aqua/AppOpts.scala:124]: Let's move it to Prelude right now?

File: cli/src/main/scala/aqua/AppOpts.scala [hunk 11]
```diff
@@ -193,7 +198,16 @@ object AppOpts {
...
+      .flag(
+        "scheduled",
+        "Generate air code for script storage. Without error handling wrappers and hops on relay. Will ignore other options"
+      )
+      .map(_ => true)
+      .withDefault(false)
+
+  val noGlobalImports: Opts[Boolean] =
+    Opts
+      .flag("no-global-imports", "Don't use default aqua imports, i. e. builtin.aqua")
```
Comment (alari) [inline cli/src/main/scala/aqua/AppOpts.scala:210]: It has nothing about `builtin.aqua`

**Review by alari — APPROVED** [review]

File: cli/src/main/scala/aqua/AppOpts.scala [hunk 12]
```diff
@@ -193,7 +198,21 @@ object AppOpts {
...
+        "Generate air code for script storage. Without error handling wrappers and hops on relay. Will ignore other options"
+      )
+      .map(_ => true)
+      .withDefault(false)
+
+  val noGlobalImports: Opts[Boolean] =
+    }
+
+Opts
+    .flag("no-global-imports", "Don't use default aqua import paths")
```
Comment (folex) [inline cli/src/main/scala/aqua/AppOpts.scala:212]: It's usually called "prelude" in languages. Maybe rename to `--no-prelude`?

File: npm/utils.js [hunk 13]
```diff
@@ -0,0 +1,2 @@
+// It should work in scala as js.`import`.meta.url, but it doesn't compile for some reasons
```
Comment (folex) [inline npm/utils.js:1]: maybe rename to `npm/metaUrl.js`?

File: cli/src/main/scala/aqua/AppOpts.scala [hunk 14]
```diff
@@ -193,7 +184,16 @@ object AppOpts {
...
+      .flag(
+        "scheduled",
+        "Generate air code for script storage. Without error handling wrappers and hops on relay. Will ignore other options"
+      )
+      .map(_ => true)
+      .withDefault(false)
+
+  val noPrelude: Opts[Boolean] =
+    Opts
+      .flag("no-prelude", "Don't use default aqua import paths")
```
Comment (folex) [inline cli/src/main/scala/aqua/AppOpts.scala:196]: Let's remove that flag altogether. It seems that there's no use-case for it.

There's a consideration, however, about priorities between local `aqua-lib` and a global one. Local `aqua-lib` (from node_modules in the current dir) should always be used when available, and global one ignored.

In other words, we need to make sure that the following scenario works:
1. Install global aqua via `npm i -g`
2. Install local aqua via `npm --save-dev`, so it is in package.json
3. Install local `aqua-lib@v0.2.0` via `npm --save-dev @fluencelabs/aqua-lib@v0.2.0`, so it is in package.json
4. Write simple aqua script that uses `Peer.timeout` from `@fluencelabs/aqua-lib`
5. Compile with global aqua, should fail cuz `Peer.timeout` isn't available in `0.2.0`
6. Compile with local aqua, should fail as well

In this scenario, we check that local aqua-lib of version 0.2.0 takes precedence of global aqua-lib of version 0.2.1. Usually, it's gonna be the other way around: global aqua-lib would be older than the local one.

**Pull request merged**